### ISSUE TITLE
Datepicker sometimes closes itself when selecting a year

### ISSUE THREAD

**Issue by alcalyn** [chrisdavies/tiny-date-picker#31]
When selecting a year, the Datepicker closes itself.

Context:
- Using Chrome
- `dp-below` mode
- having date input empty
- years from 1900 to today

Open year selector, it starts from 1900, then select 1905: the datepicker closes itself.

This line is called:

https://github.com/chrisdavies/tiny-date-picker/blob/master/tiny-date-picker.js#L148

`document.activeElement` contains my `<body>` element when it closes. When it does not close (expected behaviour), it contains `<a tabindex="-1" href="javascript:;" class="dp-day dp-current" data-date="-2145920400000">1</a>`

I couldn't reproduce it on demo page.

**Comment by chrisdavies** [comment]
Interesting. Yeah. There are lots of edge cases that are tricky to find. Thanks for reporting it. 

I'll get it fixed!

**Comment by alcalyn** [comment]
I can reproduce the bug on example page:

Add momentjs:

``` html
<script src="https://momentjs.com/downloads/moment-with-locales.js"></script>
```

Add options in first datepicker:

``` js
      TinyDatePicker(document.querySelector('.modal-txt'), {
            min: moment('01/01/1900', 'DD/MM/YYYY'),
            max: moment().subtract(13, 'years'),
            format: function (date) {
                return moment(date).format('DD/MM/YYYY');
            },
            parse: function (str) {
                return moment(str, 'DD/MM/YYYY').toDate();
            },
            months: [
                'Janv',
                'Févr',
                'Mars',
                'Avril',
                'Mai',
                'Juin',
                'Juil',
                'Août',
                'Sept',
                'Oct',
                'Nov',
                'Déc'
            ],
            days: [
                'Dim',
                'Lun',
                'Mar',
                'Mer',
                'Jeu',
                'Ven',
                'Sam'
            ],
            today: 'Aujourd\'hui',
            clear: 'Effacer',
            close: 'Fermer',
            mode: 'dp-below',
            weekStartsMonday: true
        });
```

Then go to the test page, and select `1905` in the year menu: the datepicker is now closed.

I have Chrome Version 58.0.3029.81 (64-bit), Ubuntu 16.04.

**Comment by alcalyn** [comment]
Well, in fact, it happens on all years that start with a Sunday.

Then `weekStartsMonday` ignore the first day of the month, and starts from 2.

See January 1905, January 1984...

**Comment by chrisdavies** [comment]
Hey, thanks for the follow up. I've been slacking off on this, but I will
be able to take a look tomorrow.

On May 10, 2017 12:01 PM, "Julien Maulny" <notifications@github.com> wrote:

Well, in fact, it happens on all years that start with a Sunday.

Then weekStartsMonday ignore the first day of the month, and starts from 2.

See January 1905, January 1984...

—
You are receiving this because you commented.

Reply to this email directly, view it on GitHub
<https://github.com/chrisdavies/tiny-date-picker/issues/31#issuecomment-300530054>,
or mute the thread

**Comment by chrisdavies** [comment]
Ah. I see. There's a further bug in that the 1st of January doesn't even display in January on those years. You have to go to December of the previous year in order to see it.

**Comment by alcalyn** [comment]
Yep, and this bug causes this one, which makes the datepicker closes when trying to selecting one of these months on Chrome

**Comment by chrisdavies** [comment]
Fixed in tiny-date-picker@2.0.9

Thanks again!
